spec/services/course/assessment/answer/multiple_response_auto_grading_service_spec.rb

```diff
@@ -21,55 +21,91 @@
 
     describe '#grade' do
       context 'when the question is requires all correct options' do
-        context 'when the correct answer is given' do
-          let(:answer_traits) { :correct }
+        context 'when only the correct answer is given' do
+          let(:answer_traits) { :with_all_correct_options }
 
           it 'marks the answer correct' do
             subject.grade(grading)
             expect(answer.grade).to eq(question.maximum_grade)
             expect(answer).to be_correct
-            expect(grading.result['messages']).to \
-              contain_exactly(question.options.correct.first.explanation)
+            expect(grading.result['messages']).
+              to contain_exactly(*answer.specific.options.map(&:explanation))
           end
         end
 
-        context 'when the wrong answer is given' do
-          let(:answer_traits) { :wrong }
+        context 'when only the wrong answer is given' do
+          let(:answer_traits) { :with_all_wrong_options }
 
           it 'marks the answer wrong' do
             subject.grade(grading)
             expect(answer).not_to be_correct
             expect(answer.grade).to eq(0)
-            expect(grading.result['messages']).to \
-              contain_exactly(answer.specific.options.first.explanation)
+            expect(grading.result['messages']).
+              to contain_exactly(*answer.specific.options.map(&:explanation))
+          end
+        end
+
+        context 'when the wrong and right answer is given' do
```

Comment:
"Answers" should be plural. Maybe "when both wrong and right answers are given"?